packages/frontend-v2/components/Plan/AddPlanModal.tsx

```diff
@@ -0,0 +1,170 @@
+      return;
+    }
+
+    // refresh the cache and close the modal
+    mutate(USE_STUDENT_WITH_PLANS_SWR_KEY);
+    resetFormAndCloseModal(createdPlanId);
+  };
+
+  const resetFormAndCloseModal = (createdPlanId?: number) => {
+    reset();
+    onClose(createdPlanId);
+  };
+
+  return (
+    <>
+      <Modal isOpen={isOpen} onClose={() => resetFormAndCloseModal()} size="md">
+        <ModalOverlay />
+        <ModalContent>
+          <form onSubmit={handleSubmit(onSubmitHandler)}>
+            <ModalCloseButton />
+            <ModalHeader color="primary.red.main" fontSize="2xl">
+              New Plan
+            </ModalHeader>
+            <ModalBody>
+              <FormControl isInvalid={errors.name != null} mb="sm">
+                <AddPlanLabel label="Title" />
+                <Input
+                  type="text"
+                  id="name"
+                  placeholder="My Plan"
+                  {...register("name", {
+                    required: "Title is required",
+                  })}
+                />
+                <FormErrorMessage>{errors.name?.message}</FormErrorMessage>
+              </FormControl>
+              <FormControl isInvalid={errors.catalogYear != null} mb="sm">
+                <AddPlanLabel label="Catalog Year" />
+                <Select
+                  id="catalogYear"
+                  placeholder="Select a Catalog Year "
+                  {...register("catalogYear", {
+                    required: "Catalog year is required",
+                    valueAsNumber: true,
+                  })}
+                >
+                  {Array.from(SUPPORTED_MAJORS.keys()).map((catalogYear) => (
+                    <option key={catalogYear} value={catalogYear}>
+                      {catalogYear}
+                    </option>
+                  ))}
+                </Select>
+                <FormErrorMessage>
+                  {errors.catalogYear?.message}
+                </FormErrorMessage>
+              </FormControl>
+              <FormControl isInvalid={errors.catalogYear != null}>
+                <AddPlanLabel label="Major" />
+                <Select
+                  id="major"
+                  placeholder="Select a Major"
+                  {...register("major", {
+                    required: "Major is required",
+                  })}
+                >
+                  {(SUPPORTED_MAJORS.get(catalogYear) ?? []).map((major) => (
+                    <option key={major} value={major}>
+                      {major}
+                    </option>
+                  ))}
+                </Select>
+                <FormErrorMessage>{errors.major?.message}</FormErrorMessage>
+              </FormControl>
+            </ModalBody>
+            <ModalFooter>
+              <Button isLoading={isSubmitting} type="submit" ml="auto">
+                Add Plan
+              </Button>
+            </ModalFooter>
+          </form>
+        </ModalContent>
+      </Modal>
```

Comment:
This seems like it might be a lot to fit in one component from a lines-of-code/readability persepective. It is all still clearly part of one unit though, so it might be okay as is.

Reply:
I totally agree, i think the only thing making me a little less inclined in doing so is that there isn't any code repeatability that we'd eliminate by separating into multiple reusable components. On the other hand, separating this into multiple presentational components will require drilling props and a lot of additional code.

Honestly, i think i may be a little lazy but since there is somewhat of a tradeoff i shall let it be haha, regardless a good point!!